components/base/kinematicbase/execution.go

```diff
@@ -501,3 +528,16 @@ func (ptgk *ptgBaseKinematics) makeCourseCorrectionGoals(
 	}
 	return goals
 }
+
+func (ptgk *ptgBaseKinematics) stepsToPlan(steps []arcStep, parentFrame string) motionplan.Plan {
+	traj := motionplan.Trajectory{}
+	path := motionplan.Path{}
+	for _, step := range steps {
+		traj = append(traj, map[string][]referenceframe.Input{ptgk.Kinematics().Name(): step.arcSegment.EndConfiguration})
+		path = append(path, map[string]*referenceframe.PoseInFrame{
+			ptgk.Kinematics().Name(): referenceframe.NewPoseInFrame(parentFrame, step.arcSegment.EndPosition),
```

Comment:
this is slightly confusing/ not immediately obvious to me...
There is not reason that `parentFrame` should not be in the `ptgk.Kinematics().Name()` frame, i.e. we do not know  that `parentFrame.Name() == ptgk.Kinematics().Name()`
This seems like an issue to me.

Reply:
The passed in parent frame is the frame used in `CurrentPosition()` from the localizer. It is from that pose, and frame, that all poses throughout the plan are calculated. So that is what is used here.

As this is a private function, we fully control all inputs to it.